docs/source/how-to-discussions-and-pull-requests.mdx

```diff
@@ -0,0 +1,151 @@
+# Interact with discussions and pull requests
+
+The `huggingface_hub` library provides a Python interface to interact with Pull Requests and Discussions on the Hub.
+Visit [the dedicated documentation page](https://huggingface.co/docs/hub/repositories-pull-requests-discussions)
+for a deeper view of what Discussions and Pull Requests on the Hub are, and how they work under the hood.
+
+## Retrieve Discussions and Pull Requests from the Hub
+
+The `HfApi` class allows you to retrieve Discussions and Pull Requests on a given repo:
+
+```python
+>>> from huggingface_hub import get_repo_discussions
+
+>>> get_repo_discussions(repo_id="bigscience/bloom-1b3")
+Pagination(
```

Comment:
The server returns a paginated response, I thought it would make sense to replicate this behavior in the lib

But you're right that pagination is only useful for the dalle-mini space (for now)

Couple of alternatives from the top of my head:
- iterate on the pagination internally and return the whole list of discussions
- return a generator / iterator that yields discussions

Reply:
I think both approaches make sense! Although the first seems nicer from user perspective, I think the iterator might be better longer term, so I would go with that :)